File: Assignment09/Q1.c
/////////////////////////////////////////////////////////////////////////////////
//
//      Include required header files
//
/////////////////////////////////////////////////////////////////////////////////
#include<stdio.h>

/////////////////////////////////////////////////////////////////////////////////
//
//      Function name:  CountEven
//      Input:          integer
//      Output:         integer
//      Description:    Accepts number from user and returns count of even digits
//      Date:           28/05/2026
//      Author:         Pratik Subhash Mahale
//
/////////////////////////////////////////////////////////////////////////////////
int CountEven(
                int iNo              // Input number
             )
{
    int iDigit = 0;                 // To store digit
    int iCount = 0;                 // To count even digits

    // Convert negative number into positive
    if(iNo < 0)
    {
        iNo = -iNo;
    }

    // Traverse all digits
    while(iNo > 0)
    {
        iDigit = iNo % 10;          // Extract digit

        // Check even digit
        if(iDigit % 2 == 0)
        {
            iCount++;
        }

        iNo = iNo / 10;             // Remove last digit
    }

    return iCount;                  // Return count
}

/////////////////////////////////////////////////////////////////////////////////
//
//      Application to test CountEven function
//
/////////////////////////////////////////////////////////////////////////////////
int main()
{
    int iValue = 0;                 // Input number
    int iRet = 0;                   // Result

    printf("Enter number\n");

    scanf("%d", &iValue);

    // Function call
    iRet = CountEven(iValue);

    // Output display
    printf("%d\n", iRet);

    return 0;                       // Successful termination
}

/////////////////////////////////////////////////////////////////////////////////
//
//      Input:      2395
//      Output:     1
//
//      Input:      1018
//      Output:     2
//
//      Input:      -1018
//      Output:     2
//
//      Input:      8462
//      Output:     4
//
/////////////////////////////////////////////////////////////////////////////////

/////////////////////////////////////////////////////////////////////////////////
//
//      Time Complexity Analysis
//
//      Let n be the number of digits in the input number
//
//      The while loop runs once for each digit
//      Each iteration performs constant time operations (O(1))
//
//      Therefore,
//      Time Complexity = O(n)
//
//      Space Complexity = O(1)
//
/////////////////////////////////////////////////////////////////////////////////

File: Assignment17/Q1.java
///////////////////////////////////////////////////////////////////
//
//      Class Name : Logic
//      Method Name: sumOfDigits
//      Input      : Integer
//      Output     : Void
//      Description: Calculate and display the sum of digits
//      Date       : 08/06/2026
//      Author     : Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

class Logic
{
    void sumOfDigits(int num)
    {
        int digit = 0;     // Stores extracted digit
        int sum = 0;       // Stores sum of digits

        // Handle negative number
        if(num < 0)
        {
            num = -num;
        }

        // Calculate sum of digits
        while(num != 0)
        {
            digit = num % 10;
            sum = sum + digit;
            num = num / 10;
        }

        System.out.println("Sum of digits is : " + sum);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Main Class
//
///////////////////////////////////////////////////////////////////

class Q1
{
    public static void main(String args[])
    {
        Logic obj = new Logic();

        obj.sumOfDigits(1234);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Input  : 1234
//
//      Output : 10
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(d)
//
//      Space Complexity : O(1)
//
//      Where d = Number of digits
//
///////////////////////////////////////////////////////////////////